### PR TITLE
feat: simple article

### PR DESCRIPTION
**Pull request by tuncaulubilge**
Native apps need the ability to load an article page with pre-populated data, similar to the ssr approach in render. This pr adds a "SimpleArticle" component to pages package, which shares most of the code with a regular article page. 

Only difference is, SimpleArticle lacks a Provider and depends on the native app to provide whole article as a json string. Alternatively, an error string can be passed to display an error state.

To simplify the codebase, iOS and Android specific code has been moved to pages package and completely shared now

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/8e8d04af2db6a5930517aec29950e96f7f32017d/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/3ed52e4c68b1eee58411c63a3b3040193c316810/index.html

File: android-app/pages/simple-article.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  onCommentsPress,
+  onCommentGuidelinesPress,
+  onLinkPress,
+  onTopicPress,
+  onVideoPress,
+  refetch
+} = NativeModules.ArticleEvents;
+const SimpleArticlePageView = SimpleArticle(config)(fetch);
+
+const platformAdConfig = {
```
[the remark this answers is not in the document]
Comment (tuncaulubilge) [reply android-app/pages/simple-article.js]: These could all be moved into pages. We kept these to allow platform specific stuff, but we haven't used any platform specifics so far. I'll move everything inside pages, it'll also force all platforms to catch up

[the remark this answers is not in the document]
Comment (tuncaulubilge) [reply android-app/pages/simple-article.js]: Moved everything to pages and made it shared. This makes "pages" package quite native specific, so we could consider renaming it to "native-pages" or something like that

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/980540154902d08baa07553f9654c9839ab41cdf/index.html

File: ios-app/pages/simple-article.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  deviceId: "",
+  deviceIdHash: "",
+  environment: "",
+  isLoggedIn: true,
+  platform: "mobile"
+};
+
+const SimpleArticlePageView = SimpleArticle(config)(fetch);
+
+const SimpleArticleView = ({ article, articleId, error }) => (
```
[the remark this answers is not in the document]
Comment (tuncaulubilge) [reply ios-app/pages/simple-article.js]: article can be null, in error cases etc, but we still need it to refetch

File: packages/pages/src/article/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import client from "../apollo-client";
+
+const { refetch: refetchArticle } = NativeModules.ArticleEvents;
+
+const ArticleWrapper = props => {
+  const { article, articleId, error } = props;
+
+  if (article || error) {
+    return (
+      <ApolloProvider client={client}>
```
Comment (craigbilner) [inline packages/pages/src/article/index.js]: would potentially abstract this out

File: android-app/pages/simple-article.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  articleId,
+  error,
+  omitErrors,
+  scale,
+  sectionName
+}) => {
+  const adConfig = { ...platformAdConfig, sectionName, testMode: adTestMode };
+
+  return (
+    <SimpleArticlePageView
```
[the remark this answers is not in the document]
Comment (aronshamash) [reply android-app/pages/simple-article.js]: Yes `ArticleWithProvider` makes more sense

[the remark this answers is not in the document]
Comment (tuncaulubilge) [reply android-app/pages/simple-article.js]: Yeah, refactored it to use the same component for both scenarios now.

**Comment by aronshamash** [comment]
LGTM but will wait until we get a green tick before approving.

**Pull request merged**